packages/ui/libs/ui/core/src/__tests__/Lists & Cards/GenericItemsList.spec.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { composeStories } from '@storybook/testing-react';
+import { fireEvent, render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import * as stories from '../../components/list/GenericItemsList/GenericItemsList.stories';
+import { GenericItemsListProps } from '../../components/list/GenericItemsList/GenericItemsList';
+
+const { Default, Title, Checkboxes, Pagination, EmptyListFallback } =
+  composeStories(stories);
+
+describe('GenericItemsList', () => {
+  it('should render default GenericItemsList', () => {
+    const { baseElement, getByText } = render(
+      <Default {...(Default.args as GenericItemsListProps<any, any>)} />
+    );
+
+    expect(baseElement).toBeInTheDocument();
+    expect(baseElement.querySelectorAll('.MuiList-root')).toHaveLength(
+      Default.args.listContainers.length + 1
+    );
+
+    expect(getByText('Container Header 1')).toBeInTheDocument();
+    expect(getByText('Container Header 2')).toBeInTheDocument();
+    expect(getByText('Container Header 3')).toBeInTheDocument();
+    expect(getByText('Item A1')).toBeInTheDocument();
+    expect(getByText('Item A2')).toBeInTheDocument();
+    expect(getByText('Item A3')).toBeInTheDocument();
+  });
+
+  it('should render with title', () => {
+    const { baseElement } = render(
+      <Title {...(Title.args as GenericItemsListProps<any, any>)} />
+    );
+
+    expect(baseElement).toBeInTheDocument();
+    expect(baseElement.querySelector('h4')).toBeInTheDocument();
+    expect(baseElement.querySelector('h4')).toHaveTextContent(
+      Title.args.listTitle
+    );
+  });
+
+  it('should render with checkboxes', () => {
+    const { baseElement, getAllByTestId } = render(
+      <Checkboxes {...(Checkboxes.args as GenericItemsListProps<any, any>)} />
+    );
+
+    const itemsCount =
+      Checkboxes.args.listContainers.reduce(
+        (acc, current) => acc + current.containerItems.length,
+        0
+      ) +
+      Checkboxes.args.listContainers.length +
+      1;
```

Comment:
some magic values. I guess that can be extracted:
```
const itemsInLists = Checkboxes.args.listContainers.reduce(
        (acc, current) => acc + current.containerItems.length,
        0
      )
```
And why is there `+1` at the end?